src/lro/src/details.rs

```diff
@@ -49,7 +49,20 @@ where
             let state = polling_policy.on_error(loop_start, attempt_count, e);
             handle_polling_error(state, operation_name)
         }
-        Ok(op) => handle_common(op),
+        Ok(op) => {
+            let (name, result) = handle_common(op);
+            match &result {
+                PollingResult::Completed(_) => (name, result),
+                PollingResult::InProgress(_) => {
+                    match polling_policy.on_in_progress(loop_start, attempt_count, &operation_name)
+                    {
+                        None => (name, result),
+                        Some(e) => (None, PollingResult::Completed(Err(e))),
+                    }
+                }
+                PollingResult::PollingError(_) => panic!(),
```

Comment:
I know this can't be hit with how the code is currently written, but panics still scare me a little. If we are going to panic, can you put a message in the panic so we know where this is coming from if it is ever hit.

Reply:
Done.  Good call.